include/inviwo/core/datastructures/geometry/decoratedmesh.h

```diff
@@ -0,0 +1,432 @@
+#else
+    template <typename T>
+    using TypeAlias = typename T::type;
+    using Vertex = std::tuple<TypeAlias<BufferTraits>...>;
+#endif
+
+    DecoratedMesh() { addBuffersImpl<0, BufferTraits...>(); }
+    DecoratedMesh(const DecoratedMesh &rhs) : Mesh(rhs) { copyConstrHelper<0, BufferTraits...>(); }
+    DecoratedMesh &operator=(const DecoratedMesh &that) {
+        if (this != &that) {
+            Mesh::operator=(that);
+            copyConstrHelper<0, BufferTraits...>();
+        }
+        return *this;
+    }
+
+    virtual DecoratedMesh *clone() const override { return new DecoratedMesh(*this); }
+
+    virtual ~DecoratedMesh() {}
+
+    void addVertices(const std::vector<Vertex> &vertices) {
+        addVerticesImpl<0, BufferTraits...>(vertices);
+    }
+
+#if _MSC_VER <= 1900
+    void addVertex(BufferTraits... args) { addVertexImpl<0>(args...); }
+    void setVertex(size_t index, BufferTraits... args) { setVertiesImpl<0>(index, args...); }
+#else
+    void addVertex(TypeAlias<BufferTraits>... args) { addVertexImpl<0>(args...); }
+
+    void setVertex(size_t index, TypeAlias<BufferTraits>... args) {
+        setVertiesImpl<0, BufferTraits...>(index, args...);
+    }
+#endif
+
+    template <typename BT>
+    void setVertex(size_t index, const typename BT::type &v) {
+        getTypedDataContainer<BT>().at(index) = v;
+    }
+
+    IndexBufferRAM *addIndexBuffer(DrawType dt, ConnectivityType ct) {
+        auto indicesRam = std::make_shared<IndexBufferRAM>();
+        auto indices = std::make_shared<IndexBuffer>(indicesRam);
+        addIndicies(Mesh::MeshInfo(dt, ct), indices);
+        return indicesRam.get();
+    }
+
+    template <unsigned I>
+    auto getTypedBuffer() {
+        return std::get<I>(buffers_).buffer_;
+    }
+
+    template <typename BT>
+    auto getTypedBuffer() {
+        return getTypedBuffer<buffertraits::detail::get_index<BT, BufferTraits...>::value>();
+    }
+
+    template <unsigned I>
+    auto getTypedBuffer() const {
+        return std::get<I>(buffers_).buffer_;
+    }
+
+    template <typename BT>
+    auto getTypedBuffer() const {
+        return getTypedBuffer<buffertraits::detail::get_index<BT, BufferTraits...>::value>();
+    }
+
+    template <unsigned I>
+    auto getTypedEditableRAMRepresentation() {
+        return getTypedBuffer<I>()->getEditableRAMRepresentation();
+    }
+
+    template <typename BT>
+    auto getEditableRAMRepresentation() {
+        return getEditableRAMRepresentation<
+            buffertraits::detail::get_index<BT, BufferTraits...>::value>();
+    }
+
+    template <unsigned I>
+    auto getTypedRAMRepresentation() const {
+        return getTypedBuffer<I>()->getRAMRepresentation();
+    }
+
+    template <typename BT>
+    auto getTypedRAMRepresentation() {
+        return getTypedRAMRepresentation<
+            buffertraits::detail::get_index<BT, BufferTraits...>::value>();
+    }
+
+    template <unsigned I>
+    auto &getTypedDataContainer() {
```

Comment:
Should be getTypedEditableDataContainer since it uses getTypedEditableRAMRepresentation
Could alsoe be called getEditableTypedBufferContainer